### PR TITLE
Use libphonenumber

### LINKED ISSUE #1273: Use libphonenumber for phone number validation
We do postal address validation but currently do nothing to validate phone numbers.

Here's a Python port of `libphonenumbers`: https://github.com/daviddrysdale/python-phonenumbers

Here's a library that implements a Django field: https://github.com/stefanfoulis/django-phonenumber-field

### PR DESCRIPTION
**Pull request by akjanik**
I want to merge this change because...

Fixes #1273

### Pull Request Checklist

(Please keep this section. It will make maintainer's life easier.)

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -66,6 +67,7 @@ oauthlib==2.0.4           # via requests-oauthlib, social-auth-core
 olefile==0.44             # via pillow
 pbr==3.1.1                # via mock
 pdfrw==0.4                # via weasyprint
+phonenumberslite==8.8.5   # via django-phonenumber-field
```
Comment (patrys) [inline requirements.txt]: We should use `phonenumbers` instead.

Comment (patrys) [reply requirements.txt]: Actually we should compare their functionality first, maybe we don't use any of the extra features provided by the full library.

Comment (akjanik) [reply requirements.txt]: Well, I put `django-phonenumber-field` and then run `pip-compile --output-file requirements.txt requirements.in` as said in `.in` file.

Comment (patrys) [reply requirements.txt]: I know. It changes its requirements based on what the project already uses. There are two versions of the `phonenumbers` library. If the full one is not found, the package will pull in the `lite` version.

Comment (akjanik) [reply requirements.txt]: > The phonenumberslite version of the package does not include the geocoding, carrier and timezone metadata (...)
https://github.com/daviddrysdale/python-phonenumbers
I don't think we use those features here.

File: saleor/userprofile/migrations/0014_auto_20171121_0430.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    # We get the model from the versioned app registry;
+    # if we directly import it, it'll be the wrong version
+    Address = apps.get_model("userprofile", "Address")
+    db_alias = schema_editor.connection.alias
+    for address in Address.objects.all():
+        if (address.country,) in REVERSED_COUNTRY_CODE:
+            phone_number = '+{}{}'.format(
+                           REVERSED_COUNTRY_CODE[(address.country,)],
+                           str(random.randint(10**9, 10**10 - 1)))
+            address.phone = phone_number
```
[the remark this answers is not in the document]
Comment (patrys) [reply saleor/userprofile/migrations/0014_auto_20171121_0430.py]: Why are we replacing people's phone numbers? My request was to do something akin to the following:

```python
try:
    number = phonenumbers.parse(address.phone, address.country)
    address.phone = phonenumbers.format_number(
        number,
        phonenumbers.PhoneNumberFormat.INTERNATIONAL)
except Exception:
    pass
```

[the remark this answers is not in the document]
Comment (akjanik) [reply saleor/userprofile/migrations/0014_auto_20171121_0430.py]: Yep, I misunderstood, working on it.

File: saleor/userprofile/migrations/0013_auto_20171120_0521.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import phonenumber_field.modelfields
+import phonenumbers
+
+
+def convert_phone_number_to_phonenumberfield(apps, schema_editor):
+    Address = apps.get_model("userprofile", "Address")
+    for address in Address.objects.all():
+        if address.phone:
+            phone_number = phonenumbers.parse(address.phone.raw_input,
+                                              address.country.code)
```
Comment (patrys) [inline saleor/userprofile/migrations/0013_auto_20171120_0521.py]: Please fold code after the opening parenthesis.

File: saleor/userprofile/migrations/0013_auto_20171120_0521.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            address.phone = phone_number
+            address.save()
+
+
+def convert_phone_number_to_str(apps, schema_editor):
+    Address = apps.get_model("userprofile", "Address")
+    for address in Address.objects.all():
+        if address.phone:
+            phone_number = str(address.phone)
+            address.phone = phone_number
```
Comment (patrys) [inline saleor/userprofile/migrations/0013_auto_20171120_0521.py]: This does not do what was requested. Please see previous review comments for an example.

**Comment by patrys** [comment]
The original comment ticks "The changes are documented" but I don't see any related changes made to our i18n documentation.

**Pull request merged**